_posts/en/newsletters/2020-07-08-newsletter.md

```diff
@@ -0,0 +1,227 @@
+
+- [Bitcoin Core #19204][] eliminates a source of wasted bandwidth during
+  Initial Block Download (IBD).  When Bitcoin Core is in IBD, it often
+  doesn't have the necessary information to validate currently
+  unconfirmed transactions, so it ignores relayed transaction
+  announcements.  However, those announcements still consume the
+  bandwidth of the receiving IBD node and its sending peers.  This PR
+  uses the [BIP133][] `feefilter` P2P message to tell peers that the
+  node doesn't want to receive announcements of any transactions that
+  pay less than 21 million BTC in fees per 1,000 vbytes, which should
+  prevent it from receiving any legitimate transaction announcements.
+  When the node finishes catching up to the tip of the best block chain,
+  it sends another `feefilter` message with its actual minimum relay
+  feerate so that it can start to receive newly-announced transactions.
+
+- [Bitcoin Core #19215][] creates [PSBTs][topic psbt] containing each
+  previous transaction that created one of the UTXOs being spent in the
+  current transaction---even for segwit v0 UTXOs.  Before this change,
+  previous transactions were only included for legacy (non-segwit)
+  UTXOs.  This change is a response to some hardware wallets now
+  requiring or recommending access to previous transactions for segwit
+  UTXOs in order to mitigate the [fee overpayment attack][].  If segwit
+  v1 ([taproot][topic taproot]) is adopted, transactions spending all
+  taproot inputs should not need this extra data by default.
+
+- [C-Lightning #3775][] adds RPC methods for [PSBT][topic psbt] lifecycle
+  management backed by C-Lightning's internal wallet. The `reserveinputs` RPC
+  method creates a PSBT by choosing UTXOs from the internal wallet as inputs to
+  satisfy the user-specified list of outputs, marking chosen UTXOs as reserved.
+  The resulting PSBT may either be supplied to the `unreserveinputs` RPC method
+  to manually release the reserved UTXOs, or to the `signpsbt` RPC method to add
+  signatures from the internal wallet. Finally, the `sendpsbt` RPC method will
+  convert a fully-signed PSBT into a ready-to-broadcast transaction, and
+  broadcast it to the network. Users should note that restarting C-Lightning
+  effectively un-reserves all previously reserved UTXOs, requiring a new PSBT be
+  created with `reserveinputs` before `signpsbt` will accept it.
+
+- [Eclair #1427][] and [#1439][Eclair #1439] add support to Eclair for
+  effective sending of [multipath payments][topic multipath
+  payments]---payments which are split into several parts, with each
+  part routed using a different path.  These PRs split payments into up
+  to six parts by default, initially allocating 0.00015 BTC to each part
+  but increasing the value of each part by a semi-random amount until
+  the full payment amount has been allocated.  Once the amounts have all
+  been selected, all of the payments parts are sent.  This is efficient but,
```

Comment:
I think the current punctuation is correct.  Normally, you'd be correct that "but" introduces an independent clause and should be preceded by a comma, but in this case the parenthetical "perhaps just as importantly" also needs to be set off by commas, which would naturally lead to the sentence:

> This is efficient, but, perhaps just as importantly, the opportunistic value increasing function may also [...]

That's ugly, so I was taught to remove the comma introducing the independent clause in that case.  Admittedly, what I probably should've done rather than that hack was just to rewrite the sentence---which I'm happy to do if you're not happy with it as-is.

Reply:
> > This is efficient, but, perhaps just as importantly, the opportunistic value increasing function may also [...]

I agree -- this is correct. If I had to rank the solutions, it would be this (or rewriting) first, followed by "but ,", and ", but" last... but it's just one opinion.